flight/PiOS/inc/pios_can.h

```diff
@@ -35,6 +35,7 @@
 
 //! The set of CAN messages
 enum pios_can_messages {
+	PIOS_CAN_ERROR = -1,
```

Comment:
So what I take from that is if you assigned a negative value to an enum type with only positive valued members, the behaviour is undefined. GCC should really warn you about that as well.